plugins/quetz_conda_suggest/quetz_conda_suggest/main.py

```diff
@@ -0,0 +1,34 @@
+import quetz, json
+from .api import router
+from quetz.deps import get_db
+from . import db_models
+from contextlib import contextmanager
+
+get_db = contextmanager(get_db)
+
+@quetz.hookimpl
+def register_router():
+    return router
+
+@quetz.hookimpl
+def post_add_package_version(version, condainfo):
+    suggest_map = {}
+    files_listing = condainfo.files
+
+    for each_file in files_listing:
+        if each_file.startswith(b"bin/"):
+            command = each_file.split(b"bin/")[1].strip().decode("utf-8").split("/")[0]
+            package = condainfo.info["name"]
+            if command not in suggest_map:
+                suggest_map[command] = package
+
+    with get_db() as db:
+        if not version.files:
+            metadata = db_models.CondaSuggestMetadata(
+                version_id=version.id, data=json.dumps(suggest_map)
+            )
+            db.add(metadata)
+        else:
+            metadata = db.query(db_models.CondaSuggestMetadata).get(version.id)
+            metadata.data = json.dumps(suggest_map)
+        db.commit()
```

Comment:
Looks all really good!

Can you write an updated conda-suggest _file_ here? And then, in the get_conda_suggest function above you can just return that file (or it can be handled like a static file).
We do that for repodata.json.

Reply:
This means that we essentially need to maintain a global state for the `channel_suggest_map` in `api.py`. Are you suggesting to maintain that, we create a file (serialization of `channel_suggest_map`) that we update every time a new package is uploaded?

I have one question though --  simple appending it is easy, but we need to maintain the constraint that the file should be sorted always (first by `keys`, then by `values`). So the update could be tricky.

Though, I do agree that incremental update is better than generating a new file always (by scanning all packages) as it's currently done.

Reply:
Also, in that case, would we require `db_models.py` at all? I don't see it's use if we serialize the global state of `channel_suggest_map` in a file.

Reply:
yeah because we could use the database everytime we regenerate the file. but it's true, we wouldn't strictly need it.

Reply:
Done!